core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/provision/clusterproxy/ClusterProxyService.java

```diff
@@ -0,0 +1,80 @@
+package com.sequenceiq.cloudbreak.core.flow2.cluster.provision.clusterproxy;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import com.google.gson.Gson;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.service.secret.vault.VaultSecret;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+
+@Component
+public class ClusterProxyService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterProxyService.class);
+
+    private static final Gson GSON = new Gson();
+
+    private StackService stackService;
+
+    private RestTemplate restTemplate;
+
+    @Value("${clusterProxy.url:}")
+    private String clusterProxyUrl;
+
+    @Value("${clusterProxy.registerConfigPath:}")
+    private String registerConfigPath;
+
+    @Autowired
+    ClusterProxyService(StackService stackService, RestTemplate restTemplate) {
+        this.stackService = stackService;
+        this.restTemplate = restTemplate;
+    }
+
+    public ConfigRegistrationResponse registerProxyConfiguration(Long stackId) {
+        Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+        ConfigRegistrationRequest proxyConfigRequest = createProxyConfigRequest(stack);
+        LOGGER.debug("Cluster Proxy config request: {}", proxyConfigRequest);
+        ResponseEntity<ConfigRegistrationResponse> response = restTemplate.postForEntity(clusterProxyUrl + registerConfigPath,
+                GSON.toJson(proxyConfigRequest), ConfigRegistrationResponse.class);
```

Comment:
Is there a specific reason for GSON usage? If not, we have a writeValueAsString and writeValueAsStringSilent method in JsonUtil class, it uses jackson inside. Same for 
"return GSON.fromJson(vaultSecretJsonString, VaultSecret.class).getPath();"

Reply:
Have switched to JsonUtil.